Frends.FTP.ListFiles/Frends.FTP.ListFiles/ListFiles.cs

```diff
@@ -0,0 +1,105 @@
+    /// Files task.
+    /// </summary>
+    public class FTP
+    {
+        /// <summary>
+        /// Gets the list of files from the FTP(S) source.
+        /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.FTP.ListFiles)
+        /// </summary>
+        /// <param name="input">Input parameters</param>
+        /// <param name="cancellationToken">Token to stop task. This is generated by Frends.</param>
+        /// <returns>List { DateTime Modified, string Name, string FullPath, long Size, bool Success }</returns>
+        public static async Task<Result> ListFiles([PropertyTab] Input input, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(input.Host)) throw new Exception("Host required.");
+            if (!string.IsNullOrWhiteSpace(input.User) && string.IsNullOrWhiteSpace(input.Password) || string.IsNullOrWhiteSpace(input.User) && !string.IsNullOrWhiteSpace(input.Password)) throw new Exception("Credentials required.");
+
+            return new Result(await GetListingAsync(input, cancellationToken));
+        }
+
+        private static async Task<List<ListObject>> GetListingAsync(Input input, CancellationToken cancellationToken)
+        {
+            var result = new List<ListObject>();
+
+            try
+            {
+                var client = new FtpClient(input.Host, input.Port, input.User, input.Password);
+                client.AutoConnect();
+
+                foreach (var item in await client.GetListingAsync(input.Path, FtpListOption.Auto, cancellationToken))
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    if (item.Type == FtpFileSystemObjectType.Directory)
+                        continue;
+
+                    if (item.Type == FtpFileSystemObjectType.File && !string.IsNullOrWhiteSpace(input.Filename) && FileMatchesMask(item.Name, input.Filename))
+                    {
+                        result.Add(new ListObject
+                        {
+                            Success = true,
+                            Modified = item.Modified,
+                            Name = item.Name,
+                            FullPath = item.FullName,
+                            Size = item.Size
+                        });
+                        continue;
+                    }
+
+                    if (item.Type == FtpFileSystemObjectType.File && string.IsNullOrWhiteSpace(input.Filename))
+                    {
+                        result.Add(new ListObject
+                        {
+                            Success = true,
+                            Modified = item.Modified,
+                            Name = item.Name,
+                            FullPath = item.FullName,
+                            Size = item.Size
+                        });
+                    }
+                }
+
+                if (result.Count < 1) { result.Add(new ListObject { Success = false }); };
```

Comment:
As previously noted - no need for this, just return an empty list and remove the success property altogether.